Xkcd-Comics/ViewModels/ComicsViewModel.swift

```diff
@@ -0,0 +1,150 @@
+        self.fetchingService = fetchingService
+        self.parsingService = parsingService
+        fetchComic()
+    }
+    
+    func fetchComic() {
+        fetchingService.fetchComic(fetchingURL: .latestComic) { result in
+            switch result {
+            case .success(let comic):
+                DispatchQueue.main.async {
+                    self.comic = comic
+                    self.checkIfSaved(comic: comic)
+                }
+            case .failure(let error):
+                print(error.localizedDescription)
+            }
+        }
+    }
+    
+    func fetchPreviousComic() {
+        print("VM fetchPreviousComic: number of current comic: \(comic.num)")
+        fetchingService.fetchComic(fetchingURL: .specificComic(comic.num - 1)) { result in
+            switch result {
+            case .success(let comic):
+                DispatchQueue.main.async {
+                    self.comic = comic
+                    self.checkIfSaved(comic: comic)
+                }
+            case .failure(let error):
+                DispatchQueue.main.async {
+                    self.isShowingPreviousComicAlert = true
+                }
+                print("ComicsVM, fetching previous comic failed. Error: \(error)")
+            }
+        }
+    }
+    
+    func fetchNextComic() {
+        fetchingService.fetchComic(fetchingURL: .specificComic(comic.num + 1)) { result in
+            switch result {
+            case .success(let comic):
+                DispatchQueue.main.async {
+                    self.comic = comic
+                    self.checkIfSaved(comic: comic)
+                }
+            case .failure(let error):
+                DispatchQueue.main.async {
+                    self.isShowingNextComicAlert = true
+                }
+                print("ComicsVM, fetching next comic failed. Error: \(error)")
+            }
+        }
+    }
+    
+    func toggleDescription() {
+        isShowingExplanation.toggle()
+    }
+    
+    func toggleInfo() {
+        isShowingInfo.toggle()
+    }
+    
+    func searchComic(searchNum: String) {
+        if isShowingSearch {
+            // TODO: error warning if the input is not int
+            fetchingService.fetchComic(fetchingURL: .specificComic(Int(searchNum) ?? 1)) { result in
+                switch result {
+                case .success(let comic):
+                    DispatchQueue.main.async {
+                        self.comic = comic
+                        self.isShowingSearch = false
+                        self.checkIfSaved(comic: comic)
+                    }
+                case .failure(let error):
+                    print("ComicsVM, searching next comic failed. Error: \(error)")
+                }
+            }
+        } else {
+            isShowingSearch = true
+        }
+    }
+    
+    func saveAsFavourite(comic: Comic) {
+        if comics.contains(comic) {
```

Comment:
[Minor] Style: Early returns with guards should be preferred rather than if/else, in all situations where we have a main thing we want to do, but need to just check specific cases first. Increases readability.

```swift
guard !comics.contains(comic) else {
    return
}
```